crates/wasi-provider/src/states/initializing.rs

```diff
@@ -0,0 +1,83 @@
+use std::collections::HashMap;
+
+use log::info;
+
+use crate::PodState;
+use kubelet::container::{ContainerKey, Status as ContainerStatus};
+use kubelet::state::prelude::*;
+
+use super::error::Error;
+use super::starting::{start_container, ContainerHandleMap, Starting};
+
+#[derive(Debug)]
+pub struct Initializing;
+
+#[async_trait::async_trait]
+impl State<PodState> for Initializing {
+    async fn next(
+        &self,
+        pod_state: &mut PodState,
+        pod: &Pod,
+    ) -> anyhow::Result<Transition<PodState>> {
+        let mut container_handles: ContainerHandleMap = HashMap::new();
+
+        // TODO: init containers need to run in alphabetical order. We'll have to sort by container name.
+        for init_container in pod.init_containers() {
+            info!(
+                "Starting init container {:?} for pod {:?}",
+                init_container.name(),
+                pod.name()
+            );
+
+            let handle = start_container(pod_state, pod, &init_container).await?;
+
+            while let Some((_name, status)) = pod_state.run_context.status_recv.recv().await {
+                // TODO: implement a container state machine such that it will self-update the Kubernetes API as it transitions through these stages.
+                //let s = serde_json::json!({
+                //    "metadata": {
+                //        "resourceVersion": "",
+                //    },
+                //    "status": {
+                //        "containerStatuses": [
```

Comment:
This will need to be `initContainerStatuses` when you uncomment

Reply:
Nice catch.